jina/drivers/convert.py

```diff
@@ -101,17 +101,16 @@ def convert(self, d):
         d.blob.CopyFrom(array2pb(np.frombuffer(d.buffer)))
 
 
-class Blob2PngURI(BaseConvertDriver):
+class NdArray2PngURI(BaseConvertDriver):
     """Simple DocCrafter used in :command:`jina hello-world`,
-        it reads ``buffer`` into base64 png and stored in ``uri``"""
+        it reads ``NdArray`` into base64 png and stored in ``uri``"""
 
     def __init__(self, target='uri', width: int = 28, height: int = 28, *args, **kwargs):
         super().__init__(target, *args, **kwargs)
         self.width = width
         self.height = height
 
-    def convert(self, d):
-        arr = pb2array(d.blob)
+    def convert(self, arr):
         pixels = []
```

Comment:
Perhaps put everything into a separate function and only do here
```
arr.uri = self.my_new_function(arr)
```

Reply:
done